### ISSUE TITLE
Use http-proxy for websocket 

### ISSUE THREAD

**Issue by mburger81** [versatica/JsSIP#425]
We use your lib to make a call to an webrtc rtsp stream.
We need to configure a proxy in dev mode for your WebSocket calls, is this possible in your lib?
Doing something like that and configure a proxy in our webpack/angular2 environment does not work

var socket = new JsSIP.WebSocketInterface(/wsapi/webrtc?.....);

If that is not possible can you include this feature request for it?

**Comment by ibc** [comment]
Sorry, no idea about how Angular handles proxy stuff. Neither I understand what you try to achieve by setting such a rare URL (which does not even have wss schema).

You can create your own socket by inheriting from the JsSIP.Socket (same as the JsSIP.WebSocketInterface does) and add there your own stuff.

Closing the issue as it does not provide enough info to handle it. Dont hesitase in adding much more if possible.

**Comment by mburger81** [comment]
For example our websocket for webrtc is under CORS control, so we need an PROXY which is delivered in most frameworks with node http-proxy as fo angular2 and webpack-dev-server

So you can configure in development environment a proxy which is listen for WEBSOCKETS for example for

JsSip.WebSocketInterface('wasapi/webrtc?....') we can configure a proxy like that

```
proxy : [{
         path: /wsapi, 
         target: ws://192.168.11/,
         ws: true
}]
```

So you can handle simple CORS problems if server is not on same domain as dev environment or if you have to handle different server you can handle this by configuration tools

I think this is a basic functionality as PROXY on simple HTTP request.

**Comment by ibc** [comment]
I don't know about proxy auth stuff, so I don't understand what such a `proxy` object above means.

However, JsSIP comes with a `Socket` class to build transport classes, and a built-in `WebSocketInterface` subclass for browser WebSockets. You can create your own and add the proxy auth stuff into it. Once it works, you can send us a PR.

* http://jssip.net/documentation/3.0.x/api/socket/
* http://jssip.net/documentation/3.0.x/api/websocket/
* https://github.com/versatica/JsSIP/blob/master/lib/Socket.js
* https://github.com/versatica/JsSIP/blob/master/lib/WebSocketInterface.js

**Comment by jmillan** [comment]
@mburger81,

You are describing your scenario with many dependencies (webpack, angular, webpack-dev-server..). I cannot understand your needs based on your description.

Can you rephrase it only in terms of WebSocket connection?

**Comment by d-levin** [comment]
@mburger81 Did you ever resolve this? I think I'm having a similar problem.

**Comment by ibc** [comment]
> @mburger81 Did you ever resolve this? I think I'm having a similar problem.

So may be you want to clarify what this is about. too many dependencies, frameworks, etc here.

**Comment by d-levin** [comment]
I'm unfortunately in the same situation with respect to dependencies and frameworks. I'm trying to use JsSIP in my Vuejs project which uses nodejs, Webpack, and a dev server running on nodejs.

When I answer a call (if the dev-server serves the project) the call drops in the `session confirmed` event with the message
`Reason: Q.850;cause=27;text="DESTINATION_OUT_OF_ORDER"`

If I try it by simply adding the JsSIP library to an HTML file directly, then it works. The call is answered and kept alive until I hang up.

It's probably worth noting that I'm running a really old version (0.6.5) of JsSIP. I tried upgrading but couldn't even register using the latest version.. Basically I'm trying to integrate a legacy project into a modern JS framework.

**Comment by ibc** [comment]
If you receive "Reason: Q.850;cause=27;text="DESTINATION_OUT_OF_ORDER" that's obviously generated by your remote SIP server/gateway. You may want to check the INVITE you send to it and compre it with the working one.

And yes, you should move to latest JsSIP. If you have problems, please open a question in the mailing list (but provide detailed info, please).
